crawler/domain.go

```diff
@@ -49,24 +45,53 @@ func parseDomainsFile(data []byte) ([]Domain, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	return domains, nil
+	return domains, err
 }
 
 func (domain Domain) processAndGetNextURL(url string, wg *sync.WaitGroup, repositories chan Repository) (string, error) {
-	crawler, err := GetClientAPICrawler(domain.ClientAPI)
+	crawler, err := GetClientAPICrawler(domain.Host)
 	if err != nil {
 		return "", err
 	}
-
 	return crawler(domain, url, repositories, wg)
 }
 
 func (domain Domain) processSingleRepo(url string, repositories chan Repository) error {
-	crawler, err := GetSingleClientAPICrawler(domain.ClientAPI)
+	crawler, err := GetSingleClientAPICrawler(domain.Host)
 	if err != nil {
 		return err
 	}
-
 	return crawler(domain, url, repositories)
 }
+
+func (domain Domain) generateAPIURL(u string) (string, error) {
+	crawler, err := GetAPIURL(domain.Host)
+	if err != nil {
+		return u, err
+	}
+	return crawler(u)
+}
+
+// KnownHost checks if the host is in the domain list and return the right Domain.
+func KnownHost(link, host string, domains []Domain) (Domain, error) {
+	for _, domain := range domains {
+		if host == domain.Host {
+			// Host is found in the host list.
+			return domain, nil
+		}
+	}
+
+	// host unknown, needs to be inferred.
+	if IsGithub(link) {
+		log.Infof("%s - API inferred:%s", link, "github.com")
+		return Domain{Host: "github.com"}, nil
```

Comment:
here I think we need to return `github`, `gitlab`, ... because this is the client type to use, not the domain url

Reply:
Yes, probably is better with the domain api name.
Now it uses Domain Host when it reads from the domains.yml file, where we can have an external domain registered as "gitlab.comune.aa.it".